pkg/asset/lbconfig/lbconfig.go

```diff
@@ -0,0 +1,156 @@
+package lbconfig
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/installer/pkg/asset"
+)
+
+// LoadBalancerCategory describes the type of load balancer data.
+type LoadBalancerCategory string
+
+var (
+	// ConfigName is the name of the load balancer config file.
+	ConfigName = "openshift-lbconfig-custom-dns.yaml"
+
+	// ConfigPath is the path to the load balancer config file.
+	ConfigPath = filepath.Join("openshift", ConfigName)
+
+	// PublicLoadBalancer indicates that the data is related to public load balancers.
+	PublicLoadBalancer LoadBalancerCategory = "external-api-lb-dns-name"
+
+	// PrivateLoadBalancer indicates that the data is related to private load balancers.
+	PrivateLoadBalancer LoadBalancerCategory = "internal-api-lb-dns-name"
+)
+
+// Config generates the lbConfigForDNS ConfigMap.
+type Config struct {
+	File *asset.File
+}
+
+var _ asset.WritableAsset = (*Config)(nil)
+
+// Name returns a human friendly name for the asset.
+func (*Config) Name() string {
+	return "OpenShift Load Balancer Config"
+}
+
+// Dependencies returns all the dependencies directly needed to generate
+// the asset.
+func (*Config) Dependencies() []asset.Asset {
+	return []asset.Asset{}
+}
+
+// Generate generates the openshift-install ConfigMap.
+func (i *Config) Generate(dependencies asset.Parents) error {
+	cm, err := CreateLBConfigMap("openshift-lbconfig-custom-dns", "", "")
```

Comment:
ConfigMap name does not have to include "custom-dns" in it. Let us name it based on its contents. So, how about "openshift-lb-config" ?